### PR TITLE
Fix `TabBar` size when theme changes

### LINKED ISSUE #87511: TabContainer Floating Tabs
### Tested versions

- v4.2.stable.official [46dc27791]
- v4.1.stable.official [970459615]

### System information

Godot v4.2.stable - Windows 10.0.22000 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 SUPER (NVIDIA; 31.0.15.3713) - Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 Threads)

### Issue description

TabContainer tabs render fine in the 4.1 version of Godot I've listed above. In 4.2, they float above the tab panel and seem to shrink as much as they can around the tab text. Toggling various settings (e.g. `Clip Tabs` on the `TabContainer`) seems to fix the issue.

https://github.com/godotengine/godot/assets/4305699/a07e822a-5dd2-4a9a-b982-735426f543d3

### Steps to reproduce

- Create a tab container with a couple tabs in 4.2
- Increase font size on the tab container
- Play the project
- Remote toggle Clip Tabs on the TabContainer to see the rendering fix itself.
- Open the project in 4.1 to see the issue is gone.

### Minimal reproduction project (MRP)

[TabContainerTest.zip](https://github.com/godotengine/godot/files/14027729/TabContainerTest.zip)

### PR DESCRIPTION
**Pull request by WhalesState**
Fixes: https://github.com/godotengine/godot/issues/87511

### CODE REVIEW

File: scene/gui/tab_bar.cpp [hunk 1]
```diff
@@ -375,6 +375,7 @@ void TabBar::_notification(int p_what) {
 				_shape(i);
 			}
 
+			update_minimum_size();
 			queue_redraw();
```
Comment (YeldhamDev) [inline scene/gui/tab_bar.cpp:379]:
```suggestion
			queue_redraw();
			update_minimum_size();
```

**Review by YeldhamDev — CHANGES_REQUESTED** [review]
Should be in this order, like the other instances in the file:

**Review by YeldhamDev — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by wez470** [comment]
Thank you!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**